### PR TITLE
Printing contents of Ltac closures

### PR DESCRIPTION
**Pull request by herbelin**
A step towards #14605. For @jfehrle to comment. 

- [ ] Added / updated **test-suite**.
- [ ] Added **changelog**.
- [ ] Added / updated **documentation**.

### CODE REVIEW

**Comment by jfehrle** [comment]
This fixes the second example in #14605 using idtac, but it shows "T : <genarg:tacvalue>" in the debugger for the variable t.  But still some progress.  

image

**Comment by ppedrot** [comment]
@herbelin this is a trivial net positive change that should go in. Either you simply undraft it and we merge it, or we close it. I don't see the point of keeping this lingering around.

**Comment by ppedrot** [comment]
@herbelin you should rebase as well, to get a clean CI.

File: plugins/ltac/taccoerce.ml [hunk 1]
```diff
@@ -437,6 +419,29 @@ let pr_value env v =
...
+      Val.t Id.Map.t * (* closure *)
+      Name.t list * (* binders *)
+      Tacexpr.glob_tactic_expr (* body *)
+  | VRec of Val.t Id.Map.t ref * Tacexpr.glob_tactic_expr
+
+let (wit_tacvalue : (Empty.t, tacvalue, tacvalue) Genarg.genarg_type) =
+  let wit = Genarg.create_arg "tacvalue" in
+  let () = register_val0 wit None in
+  let () = Genprint.register_val_print0 (base_val_typ wit)
+             (function
```
Comment (ppedrot) [inline plugins/ltac/taccoerce.ml:437]: @herbelin you should probably give a name to this anonymous function, for legibility.

Comment (herbelin) [reply plugins/ltac/taccoerce.ml]: I named it "pr" (hum...).

I also rebased.

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**